tests/integration/api/network_apis_test.go

```diff
@@ -0,0 +1,167 @@
+package api_test
+
+import (
+	"fmt"
+	"net/http"
+
+	cniv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	. "github.com/onsi/ginkgo"
+	"github.com/tidwall/gjson"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/rancher/harvester/pkg/config"
+	ctlcniv1 "github.com/rancher/harvester/pkg/generated/controllers/k8s.cni.cncf.io/v1"
+	. "github.com/rancher/harvester/tests/framework/dsl"
+	"github.com/rancher/harvester/tests/framework/fuzz"
+	"github.com/rancher/harvester/tests/framework/helper"
+)
+
+const (
+	testNetworkNamespace      = "default"
+	testNetworkTypeVlan       = "L2VlanNetwork"
+	testBridgeVID             = 100
+	testNetworkBridgeTemplate = `
+{"cniVersion":"0.3.1","name":"%s","type":"bridge","bridge":"harvester-br0","promiscMode":true,"vlan":%d,"ipam":{}}
+`
+)
+
+type BridgeNetwork struct {
+	NAD *cniv1.NetworkAttachmentDefinition
+}
+
+func NewBridgeNetwork(name string, vid int) *BridgeNetwork {
+	bridgeNetwork := BridgeNetwork{
+		NAD: NewNAD(name, testNetworkTypeVlan, NewBridgeNetworkConfig(name, vid)),
+	}
+	return &bridgeNetwork
+}
+
+func NewBridgeNetworkConfig(name string, vid int) string {
+	return fmt.Sprintf(testNetworkBridgeTemplate, name, vid)
+}
+
+func NewNAD(name, networkType, config string) *cniv1.NetworkAttachmentDefinition {
+	networkLabels := map[string]string{
+		"test.harvester.cattle.io":          "harvester-test",
+		"networks.harvester.cattle.io/type": networkType,
+	}
+	return &cniv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: testNetworkNamespace,
+			Labels:    networkLabels,
+		},
+		Spec: cniv1.NetworkAttachmentDefinitionSpec{
+			Config: config,
+		},
+	}
+}
+
+var _ = Describe("verify network APIs", func() {
+
+	var (
+		scaled        *config.Scaled
+		nadController ctlcniv1.NetworkAttachmentDefinitionController
+		networkName   string
+	)
+
+	BeforeEach(func() {
+		scaled = harvester.Scaled()
+		nadController = scaled.CniFactory.K8s().V1().NetworkAttachmentDefinition()
```

Comment:
I mean we can move these out of the BeforeEach function

Reply:
If move variables  these out, the values of these variables will be assigned before all tests are executed.